### DISCUSSION TITLE
Trouble running local jekyll server for the docs folder of a private project

### DISCUSSION THREAD

**Discussion by Haf-Linger** [community/community#21730] [Other Feature Feedback, Questions, & Ideas]
<p>I have private project with the documentation sitting in the docs folder.<br>
And the project repro is connected to the local repro folder on my PC.<br>
But I am getting tired of updating the local .md files, pushing the changes and waiting till github republishes the changed content to <a href="http://github.io" rel="nofollow noopener">github.io</a>.<br>
So I would like to run a local jekyll server using my local docs folder (of course using the same theme as the final github pages site)</p>
<p>I have been unable to find doc that particulary covers this case.<br>
So if anybody can point in the right direction that would be very much appreciated.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>GitHub Pages is built on <a href="https://jekyllrb.com/" rel="nofollow noopener">Jekyll</a>, a static site generator written in Ruby. You can follow <a href="https://jekyllrb.com/docs/" rel="nofollow noopener">these instructions</a> (except for the part about generating a new Jekyll site, of course) to get it running on your system and use the <a href="https://jekyllrb.com/docs/usage/" rel="nofollow noopener"><code>jekyll serve</code> command</a> to serve the site locally. You may also need to install the gem for whatever theme your Pages site is using.</p>
<p>I hope that helps!</p>
